ChemistryLib/CreateChemicalSolverInterface.cpp

```diff
@@ -0,0 +1,137 @@
+#include "PhreeqcKernel.h"
+#include "PhreeqcKernelData/AqueousSolution.h"
+#include "PhreeqcKernelData/CreateAqueousSolution.h"
+#include "PhreeqcKernelData/CreateKineticReactant.h"
+#include "PhreeqcKernelData/ReactionRate.h"
+
+#include "ThirdParty/iphreeqc/src/src/phreeqcpp/cxxKinetics.h"
+
+namespace
+{
+std::string parseDatabasePath(BaseLib::ConfigTree const& config)
+{
+    // database
+    //! \ogs_file_param{prj__chemical_system__database}
+    auto const database = config.getConfigParameter<std::string>("database");
+    auto path_to_database =
+        BaseLib::joinPaths(BaseLib::getProjectDirectory(), database);
+
+    if (!BaseLib::IsFileExisting(path_to_database))
+    {
+        OGS_FATAL("Not found the specified thermodynamicdatabase: %s",
+                  path_to_database.c_str());
+    }
+
+    INFO("Found the specified thermodynamic database: %s",
+         path_to_database.c_str());
+
+    return path_to_database;
+}
+}  // namespace
+
+namespace ChemistryLib
+{
+template <>
+std::unique_ptr<ChemicalSolverInterface>
+createChemicalSolverInterface<ChemicalSolver::Phreeqc>(
+    MeshLib::Mesh const& mesh,
+    std::vector<std::pair<int, std::string>> const&
+        process_id_to_component_name_map,
+    BaseLib::ConfigTree const& config, std::string const& output_directory)
+{
+    auto path_to_database = parseDatabasePath(config);
+
+    // solution
+    auto aqueous_solution = PhreeqcIOData::createAqueousSolution(
+        //! \ogs_file_param{prj__chemical_system__solution}
+        config.getConfigSubtree("solution"),
+        process_id_to_component_name_map);
+
+    // kinetic reactants
+    auto kinetic_reactants = PhreeqcIOData::createKineticReactants(
+        //! \ogs_file_param{prj__chemical_system__kinetic_reactants}
+        config.getConfigSubtreeOptional("kinetic_reactants"), mesh);
+
+    // rates
+    auto reaction_rates = createReactionRates<PhreeqcIOData::ReactionRate>(
+        //! \ogs_file_param{prj__chemical_system__rates}
+        config.getConfigSubtreeOptional("rates"));
+
+    auto const num_chemical_systems = mesh.getNumberOfBaseNodes();
+    std::vector<PhreeqcIOData::AqueousSolution> aqueous_solutions(
+        num_chemical_systems, aqueous_solution);
+
+    // equilibrium phases
+    auto equilibrium_phases = PhreeqcIOData::createEquilibriumPhases(
+        //! \ogs_file_param{prj__chemical_system__equilibrium_phases}
+        config.getConfigSubtreeOptional("equilibrium_phases"), mesh);
+
+    // output
+    auto const& components = aqueous_solution.components;
+    auto const project_file_name = BaseLib::joinPaths(
+        output_directory,
+        BaseLib::extractBaseNameWithoutExtension(config.getProjectFileName()));
+    auto output = PhreeqcIOData::createOutput(
+        components, equilibrium_phases, kinetic_reactants, project_file_name);
+
+    return std::make_unique<PhreeqcIOData::PhreeqcIO>(
+        project_file_name, std::move(path_to_database),
```

Comment:
I wonder why the `project_file_name` isn't moved, too?

Reply:
Fixed.